### PR TITLE
sevenzip 22.00

### PR DESCRIPTION
**Pull request by ismail**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by ismail** [comment]
Sent a message upstream about the `sys/sysmacros.h` problem: https://sourceforge.net/p/sevenzip/discussion/45797/thread/9c2d9061ce/#01e7

File: Formula/sevenzip.rb [hunk 1]
```diff
@@ -30,6 +30,11 @@ def install
         ["gcc", "g"]
       end
 
+      if OS.mac?
+        system "sed", "-i", "''", "s/sysmacros.h/types.h/", "../../Common/FileStreams.cpp"
+        system "sed", "-i", "''", "s/sysmacros.h/types.h/", "../../UI/Common/UpdateCallback.cpp"
+      end
```
Comment (carlocab) [inline Formula/sevenzip.rb:36]: Use `inreplace` instead of shelling out to `sed`. Also, please add a comment linking to the upstream report.

**Comment by carlocab** [comment]
Let's move it out of the `cd` block to avoid having to rely on directory traversal:
```diff
diff --git a/Formula/sevenzip.rb b/Formula/sevenzip.rb
index 5c762c34b76..9e89b7c00c6 100644
--- a/Formula/sevenzip.rb
+++ b/Formula/sevenzip.rb
@@ -22,6 +22,13 @@ class Sevenzip < Formula
   end
 
   def install
+    # See https://sourceforge.net/p/sevenzip/discussion/45797/thread/9c2d9061ce/#01e7
+    if OS.mac?
+      inreplace ["Common/FileStreams.cpp", "UI/Common/UpdateCallback.cpp"].map { |d| buildpath/"CPP/7zip"/d },
+                "sysmacros.h",
+                "types.h"
+    end
+
     cd "CPP/7zip/Bundles/Alone2" do
       mac_suffix = Hardware::CPU.intel? ? "x64" : Hardware::CPU.arch
       mk_suffix, directory = if OS.mac?
```

**Comment by ismail** [comment]
> Let's move it out of the `cd` block to avoid having to rely on directory traversal:

Done, thanks for the fix!

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**